src/minErr.js

```diff
@@ -34,19 +34,32 @@ function minErr(module, ErrorConstructor) {
   ErrorConstructor = ErrorConstructor || Error;
   return function() {
     var SKIP_INDEXES = 2;
+    var DEFAULT_MAX_DEPTH = 3;
 
     var templateArgs = arguments,
+      argsDebuggedStringIndexes = [],
       code = templateArgs[0],
       message = '[' + (module ? module + ':' : '') + code + '] ',
       template = templateArgs[1],
       paramPrefix, i;
 
-    message += template.replace(/\{\d+\}/g, function(match) {
-      var index = +match.slice(1, -1),
-        shiftedIndex = index + SKIP_INDEXES;
+    function getArgumentAndConvertToDebugString(index, maxDepth) {
```

Comment:
@gkalpak  Thanks for your Feedback, 
I prefer Option(1) too but,
I would like to ask if we use option (1) how we can add tests for it ?
I mean how we can change the object max depth during the tests ?
Is there any case already implemented like this ?